### PR TITLE
enhance writeloop

### PR DESCRIPTION
**Pull request by antJack**
cancel write loop for upstream connections, still reserve for downstream connections

### Issues associated with this PR

Your PR should present related issues you want to solve.

### Sign the CLA
Make sure you have signed the [CLA](https://www.clahub.com/agreements/alipay/sofa-mosn)

### Solutions
You should show your solutions about the issues in your PR, including the overall solutions, 
details and the changes. At this time, Chinese is allowed to describe these.

### UT result
Unit Test is needed if the code is changed, your unit test should cover boundary cases, corner cases, and some exceptional cases.
And you need to show the ut result.

### Benchmark
If your code involves the processing of every request, you should give the Benchmark Result

### Code Style
+ Make sure `Goimports` has run
+ Show `Golint` result

### CODE REVIEW

File: pkg/network/connection.go [hunk 1]
```diff
@@ -94,6 +94,10 @@ type connection struct {
 	connected uint32
 	startOnce sync.Once
 	eventLoop *eventLoop
+
+	writeLock 	sync.Mutex
```
Comment (taoyuanyuan) [inline pkg/network/connection.go]: code style
